tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1763,16 +1763,8 @@ struct DynamicIotaBroadcast : public OpRewritePattern<DynamicIotaOp> {
 
     auto iotaDimension = iota.iota_dimension();
     auto iotaDimensionInt = iotaDimension;
-
-    auto convertedShape = rewriter.create<arith::IndexCastOp>(
```

Comment:
what if the `output_shape` has index type? It seems that slice op does not support index tensor as input?

Reply:
Thanks, only cast if index type.